.github/workflows/Upgrade-CI.yml

```diff
@@ -0,0 +1,285 @@
+          working-directory: integration-tests
+
+      - name: Print info about the original cluster
+        run: |
+          echo $HOME
+
+          kubectl get namespaces
+          echo "============================= event-bridge-manager ============================="
+          kubectl describe pod -n event-bridge-manager
+          echo "============================= event-bridge-operator ============================"
+          kubectl describe pod -n event-bridge-operator
+
+
+      - name: Change image names
+        run: |
+          echo "FLEET_MANAGER_CONTAINER_NAME=openbridge/fleet-manager:$NEW_BRANCH" >> $GITHUB_ENV
+          echo "FLEET_SHARD_MINIKUBE_CONTAINER_NAME=openbridge/fleet-shard:$NEW_BRANCH" >> $GITHUB_ENV
+          echo "EXECUTOR_CONTAINER_NAME=openbridge/executor:$NEW_BRANCH" >> $GITHUB_ENV
+      - name: Checkout to new version
+        uses: actions/checkout@v2
+        with:
+          ref: "${{ env.NEW_BRANCH }}"
+      - name: Build all images and resources
+        uses: ./.github/actions/maven
+        with:
+          maven-command: clean install --errors --quiet -Dquickly
+      - name: Build and Publish JVM Container - Fleet Manager
+        run: |
+          docker build -f docker/Dockerfile.jvm -t $FLEET_MANAGER_CONTAINER_NAME manager/
+          kind load docker-image $FLEET_MANAGER_CONTAINER_NAME
+      - name: Build and Publish JVM Container - Minikube Fleet Shard
+        run: |
+          docker build -f docker/Dockerfile.jvm -t $FLEET_SHARD_MINIKUBE_CONTAINER_NAME shard-operator/
+          kind load docker-image $FLEET_SHARD_MINIKUBE_CONTAINER_NAME
+      - name: Build and Publish JVM Container - Executor
+        run: |
+          docker build -f docker/Dockerfile.jvm -t $EXECUTOR_CONTAINER_NAME executor/
+          kind load docker-image $EXECUTOR_CONTAINER_NAME
+      - name: Use built Fleet manager and operator images
+        run: |
+          kustomize edit set image event-bridge-manager=$FLEET_MANAGER_CONTAINER_NAME
+          kustomize edit set image event-bridge-shard-operator=$FLEET_SHARD_MINIKUBE_CONTAINER_NAME
+        working-directory: kustomize/overlays/ci
+      - name: Use built Ingress and Executor images
+        run: |
+          sed -i -E "s|(.*EVENT_BRIDGE_EXECUTOR_IMAGE:).*|\1 $EXECUTOR_CONTAINER_NAME|" overlays/ci/shard/patches/deploy-config.yaml
+          sed -i -E "s|(.*INGRESS_OVERRIDE_HOSTNAME:).*|\1 $KIND_MAIN_NODE_HOSTNAME|" overlays/ci/shard/patches/deploy-config.yaml
+          sed -i -E "s|(.*EVENT_BRIDGE_K8S_ORCHESTRATOR:).*|\1 kind|" overlays/ci/shard/patches/deploy-config.yaml
+        working-directory: kustomize
+      - name: Set Kind hostname in the Manager configuration
```

Comment:
I think it may have sense to store overall RHOSE CI installation into a dedicated action, WDYT?
Though that may be addressed separately.

Reply:
Yes, you are right. However, it will affects also Manage CI workflow. Will do this separately in the different PR.